GeoLib/QuadTree.h

```diff
@@ -71,9 +71,9 @@ template <typename POINT> class QuadTree
 	 */
 	~QuadTree()
 	{
-		if (_is_leaf)
-			for (std::size_t k(0); k < 4; k++)
-				delete _childs[k];
+		for (std::size_t k(0); k < 4; k++) {
+			delete _childs[k];
```

Comment:
technically, it doesn't make any difference, but it'd be good if you could rename "_childs" to "_children".